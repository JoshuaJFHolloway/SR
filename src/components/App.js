import React, {Component} from 'react';
import data from './../event';
import Logo from './Logo';
import Toolbar from './Toolbar';
import SpareRoomLogo from './../assets/logo.png';
import BarPhoto from './../assets/bar-photo.jpg';

class App extends Component {
  constructor() {
    super();

    this.state = {

      results: data,
    }
  };

  render() {
    return (
      <div>
        <Logo
          logo = {SpareRoomLogo}
          alt = {"SpareRoom Logo"}
        />
        <Toolbar
          />
      </div>
    )
  }
}

export default App;