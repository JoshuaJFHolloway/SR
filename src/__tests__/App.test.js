import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';

describe('App', () => {

  let app = shallow(<App/>);

  describe('Rendering components', () => {
    it('renders correctly', () => {
      expect(app).toMatchSnapshot();
    });

    it('renders a Logo component', () => {
      expect(app.find('Logo').exists()).toBe(true);
    });

    it('renders a Toolbar component', () => {
      expect(app.find('Toolbar').exists()).toBe(true);
    });

    // describe('Events not clicked', () => {
    //   beforeEach(() => {
    //     app = shallow(<App />);
    //   });
    //   it('does not render another Toolbar component', () => {
    //     expect(app.find('Toolbar').exists()).toBe(true);
    //   });
    // });

    // describe('Events is clicked', () => {
    //   beforeEach(() => {
    //     app.setState({eventsToolbarClicked: true})
    //   });
    //
    //   it('renders renders another Toolbar component', () => {
    //     expect(app.find('Toolbar').exists()).toBe(true);
    //   });
    // });

  });

  describe('initializes state successfully', () =>{
    it('initializes results that include the event.json', () => {
      expect(app.state('results')).toHaveProperty("event.areas-covered");
    });
  });

  describe('props successfully passing down to the Logo component', () =>{
    it('passes this.state.images.logo', () => {
      expect(app.find('Logo').prop('logo'))
    });
  });

  describe('updating the state', () => {
    beforeEach(() => {
      app = shallow(<App />);
    });

    it('should update the state when events is clicked', () => {
      const event = { target: { className: 'eventsToolbarClicked', value: true } };
      app.instance().handleClickToggle(event);
      expect(app.state('params')).toEqual({"eventsNestedBarClicked": false, "eventsToolbarClicked": true});
    });

    it("should update the state when the first 'events' on the nested toolbar is clicked", () => {
      const event2 = { target: { className: 'eventsToolbarClicked', value: true } };
      app.instance().handleClickToggle(event2);
      const event = { target: { className: 'eventsNestedBarClicked', value: true } };
      app.instance().handleClickToggle(event);
      expect(app.state('params')).toEqual({"eventsNestedBarClicked": true, "eventsToolbarClicked": true});
    });

    it('should change both states to false if eventsToolbar is false', () => {
      const event1 = { target: { className: 'eventsToolbarClicked', value: true } };
      app.instance().handleClickToggle(event1);
      const event2 = { target: { className: 'eventsNestedBarClicked', value: true } };
      app.instance().handleClickToggle(event2);
      const event3 = { target: { className: 'eventsToolbarClicked', value: false } };
      app.instance().handleClickToggle(event3);
      expect(app.state('params')).toEqual({"eventsNestedBarClicked": false, "eventsToolbarClicked": false});
    })



    //
    // it('saves output state when button is clicked', () => {
    //   form.instance().handleDisplayClick();
    //   expect(form.state('savedFirstScaleScore')).toEqual(99);
    // });
    //
    // it('should change the buttonClicked state to true', () => {
    //   form = shallow(<Form />);
    //   form.instance().handleDisplayClick();
    //   expect(form.state('buttonClicked')).toEqual(true);
    // });
  });



});