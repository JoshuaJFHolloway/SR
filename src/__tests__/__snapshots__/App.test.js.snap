// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rendering components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Picture
          alt="SpareRoom Logo"
          picture="logo.png"
/>,
        <Toolbar
          className2="eventsToolbarClicked"
          eventHandler={[Function]}
          tab1="Home"
          tab2="Events"
          width="19%"
/>,
        <div
          style={
                    Object {
                              "marginTop": "10px",
                              "verticalAlign": "bottom",
                            }
          }
/>,
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "SpareRoom Logo",
          "picture": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className2": "eventsToolbarClicked",
          "eventHandler": [Function],
          "tab1": "Home",
          "tab2": "Events",
          "width": "19%",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "style": Object {
            "marginTop": "10px",
            "verticalAlign": "bottom",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Picture
            alt="SpareRoom Logo"
            picture="logo.png"
/>,
          <Toolbar
            className2="eventsToolbarClicked"
            eventHandler={[Function]}
            tab1="Home"
            tab2="Events"
            width="19%"
/>,
          <div
            style={
                        Object {
                                    "marginTop": "10px",
                                    "verticalAlign": "bottom",
                                  }
            }
/>,
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "SpareRoom Logo",
            "picture": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className2": "eventsToolbarClicked",
            "eventHandler": [Function],
            "tab1": "Home",
            "tab2": "Events",
            "width": "19%",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "style": Object {
              "marginTop": "10px",
              "verticalAlign": "bottom",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
